### PR TITLE
Feature #394: Revamp Quickstart with more information

### PR DESCRIPTION
**Pull request by osopardo1**
## Description

Adds #394 

## Type of change

Documentation Change. 

- Included Java, Maven/SBT, and Spark dependencies to install.
- Separate between Developer Environment and the Interactive one.
- Cover basic examples as [other Formats Quickstarts](https://docs.delta.io/latest/quick-start.html). 
- Separate code snippets into Python, Scala, and SQL
- Add specific instruction for Google Cloud Platform (GPC) and Amazon Web Services (AWS)
## Checklist:

Here is the list of things you should do before submitting this pull request:

- [x] Change the documentation.
- [x] Your branch is updated to the main branch (dependent changes have been merged).

## How Has This Been Tested? (Optional)

- Changed the current tests for the Quickstart in the `DocumentationTests` file. 
- Tested with Local deployment for `pyspark`.
- Tested with Local deployment for `spark-shell`.
- Tested with Local deployment for `spark-sql`.

### CODE REVIEW

**Comment by osopardo1** [comment]
This is still a Work In Progress, need to be completed. But you can see the preview [here](https://github.com/osopardo1/qbeast-spark/blob/394-revamp-quickstart/docs/Quickstart.md).

File: docs/Quickstart.md [hunk 1]
```diff
@@ -4,180 +4,392 @@
...
+You can run the qbeast-spark library in two different ways:
+
+1. Interactively: Start the Spark shell (Scala or Python) with Qbeast Spark and Delta Lake and run the code snippets.
+2. As a Project: Set up a Maven or SBT project (Scala or Java), copy the code snippets into a source file, and run the project.
+
+Before starting, ensure you have the following:
+
+- **Java 8+**: Ensure that Java is installed and properly configured.
+- **SBT/Gradle/Maven**: This is for managing dependencies if running in a development environment.
+- **Apache Spark 3.0+**: A Spark installation with support for Scala 2.12
```
Comment (cdelfosse) [inline docs/Quickstart.md:54]: Spark 3.5?

File: docs/Quickstart.md [hunk 2]
```diff
@@ -4,180 +4,392 @@
...
+
+### Download JAVA
+
+As mentioned in the official Apache Spark installation instructions [here](https://spark.apache.org/docs/latest/index.html#downloading), make sure you have a valid Java version installed (8, 11, or 17) and that Java is configured correctly on your system using either the system `PATH` or `JAVA_HOME` environmental variable.
```
Comment (cdelfosse) [inline docs/Quickstart.md:58]: Do you know if Java 21 or 22 would work too?

File: docs/Quickstart.md [hunk 3]
```diff
@@ -4,180 +4,392 @@
...
-from pyspark.sql import SparkSession
+qbeast_df = spark.read.format("qbeast").load("/tmp/qbeast_table")
 
-# Initialize Spark session with Qbeast and Delta support, running locally
-spark = SparkSession.builder \
-    .appName("QbeastExample") \
-    .master("local[*]") \
-    .config("spark.jars.packages", "io.qbeast:qbeast-spark_2.12:0.7.0,io.delta:delta-spark_2.12:3.1.0") \
-    .getOrCreate()
+qbeast_df.sample(0.3).show()
```
Comment (cdelfosse) [inline docs/Quickstart.md:299]: Shouldn't you explain what is sampling?

**Review by cdelfosse — CHANGES_REQUESTED** [review]

File: docs/Quickstart.md (hunk 1 above)
Comment (osopardo1) [reply docs/Quickstart.md:54]: True!!

File: docs/Quickstart.md (hunk 2 above)
Comment (osopardo1) [reply docs/Quickstart.md:58]: I think for[ version 4.x of Spark is build for Java 21](https://lists.apache.org/thread/66xx1k9965h440zlrhll9dbcmcbcy8no) , but 3.x still not :(

File: docs/Quickstart.md (hunk 3 above)
Comment (osopardo1) [reply docs/Quickstart.md:299]: Sure, I will come up with a few lines

Comment (osopardo1) [reply docs/Quickstart.md]: You can check the file again and let me know wdyt about the explanation 🙈

**Review by cdelfosse — APPROVED** [review]
LGTM

**Review by Jiaweihu08 — APPROVED** [review]

**Pull request merged**